servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Single.java

```diff
@@ -294,6 +295,46 @@ public final Completable flatMapCompletable(Function<? super T, ? extends Comple
         return afterFinally(doFinally);
     }
 
+    /**
+     * Invokes the corresponding method on {@code whenFinally} {@link TerminalSignalConsumer} argument when any of the
+     * following terminal methods are called:
+     * <ul>
+     *     <li>{@link Subscriber#onSuccess(Object)} - invokes {@link TerminalSignalConsumer#onSuccess(Object)}</li>
+     *     <li>{@link Subscriber#onError(Throwable)} - invokes {@link TerminalSignalConsumer#onError(Throwable)}</li>
+     *     <li>{@link Cancellable#cancel()} - invokes {@link TerminalSignalConsumer#cancel()}</li>
+     * </ul>
+     * for Subscriptions/{@link Subscriber}s of the returned {@link Single}.
+     * <p>
+     * The order in which {@code whenFinally} will be invoked relative to the above methods is undefined. If you need
+     * strict ordering see {@link #beforeFinally(TerminalSignalConsumer)} and
+     * {@link #afterFinally(TerminalSignalConsumer)}.
+     * <p>
+     * From a sequential programming point of view this method is roughly equivalent to the following:
+     * <pre>{@code
+     *  try {
+     *      T result = resultOfThisSingle();
+     *  } finally {
+     *      // NOTE: The order of operations here is not guaranteed by this method!
+     *      nextOperation(); // Maybe notifying of cancellation, or termination
+     *      doFinally.onTerminalSignal(); // Invokes the corresponding terminal signal
+     *  }
+     * }</pre>
+     * @param doFinally an appropriate method of {@link TerminalSignalConsumer} is invoked exactly once, when any of the
```

Comment:
(Same suggestion for `Publisher` and `Completable` methods)

```suggestion
     * @param doFinally For each subscribe of the returned {@link Single}, at most one method of this {@link TerminalSignalConsumer} will be invoked.
```

I think we can remove the below method mappings now as I have suggested to add them to `TerminalSignalConsumer`.